### PR TITLE
Fix ARGF 

### PR DESCRIPTION
**Pull request by buzztaiki**
Fixed `ARGF` class being `Object`.

Since `ARGF` does not have a named class, I implemented it using `ARGFClass` in reference to the `ENV` implementation.

Most of type definitions is borrowed from `io.rbs`.

References:
- https://docs.ruby-lang.org/ja/latest/class/ARGF=2eclass.html
- https://docs.ruby-lang.org/en/3.1/ARGF.html
- https://github.com/ruby/ruby/blob/7b1ece9b9490a892861f2336ae41d611a06bdf2b/io.c#L14889-L14894
- https://github.com/ruby/rbs/blob/ff384ffe425f8d38b0111ba00ead3a53d1519285/core/env.rbs#L258

### CODE REVIEW

File: core/argf.rbs [hunk 1]
```diff
@@ -0,0 +1,962 @@
...
+  #
+  #     ARGF.lineno      #=> 0
+  #     ARGF.readline    #=> "This is line 1\n"
+  #     ARGF.lineno      #=> 1
+  #     ARGF.lineno = 0  #=> 0
+  #     ARGF.lineno      #=> 0
+  #
+  %a{annotate:rdoc:copy:ARGF#lineno=}
+  # TODO: call-seq says it returns an integer, but the implementation does not
+  def lineno=: (Integer) -> untyped
```
[the remark this answers is not in the document]
Comment (soutaro) [reply core/argf.rbs:526]: @buzztaiki You may want to move the comment to the end of the line.

```suggestion
  def lineno=: (Integer) -> untyped    # TODO: call-seq says it returns an integer, but the implementation does not
```

But this is an interesting point that we sometimes want this kind of comment, which shouldn't be included in the doc. Hm...

[the remark this answers is not in the document]
Comment (buzztaiki) [reply core/argf.rbs]: Thank you for your suggestion. I applied it.

[the remark this answers is not in the document]
Comment (buzztaiki) [reply core/argf.rbs]: I filed this returning value issue to ruby's bug tracker.
https://bugs.ruby-lang.org/issues/18753

I will keep this TODO as a personal and remove it from the comments.

[the remark this answers is not in the document]
Comment (buzztaiki) [reply core/argf.rbs]: RBS comment job was passed in 31842c0b.

[the remark this answers is not in the document]
Comment (nobu) [reply core/argf.rbs]: I doubt if the return type of setter methods makes sense generally, as it does only when called with `__send__`.
Shouldn't RBS consider this language spec?

**Comment by buzztaiki** [comment]
Thanks you for your merging! I see you decided to put `ARGFClass` in `RBS::Unnamed`.

**Pull request merged**